src/Microsoft.OData.Core/UriParser/Parsers/CountSegmentParser.cs

```diff
@@ -0,0 +1,138 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="CountSegmentParser.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    using System;
+    using ODataErrorStrings = Microsoft.OData.Strings;
+
+    /// <summary>
+    /// Syntactic parser for the $count segment.
+    /// </summary>
+    internal sealed class CountSegmentParser
+    {
+        /// <summary>
+        /// Reference to the lexer.
+        /// </summary>
+        private readonly ExpressionLexer lexer;
+
+        /// <summary>
+        /// Method used to parse arguments.
+        /// </summary>
+        private readonly UriQueryExpressionParser parser;
+
+        /// <summary>
+        /// Create a new CountSegmentParser.
+        /// </summary>
+        /// <param name="lexer">Lexer positioned at a $count identifier.</param>
+        /// <param name="parser">The UriQueryExpressionParser.</param>
+        public CountSegmentParser(ExpressionLexer lexer, UriQueryExpressionParser parser)
+        {
+            ExceptionUtils.CheckArgumentNotNull(lexer, "lexer");
+            ExceptionUtils.CheckArgumentNotNull(parser, "parser");
+            this.lexer = lexer;
+            this.parser = parser;
+        }
+
+        /// <summary>
+        /// Reference to the underlying UriQueryExpressionParser.
+        /// </summary>
+        public UriQueryExpressionParser UriQueryExpressionParser
+        {
+            get { return this.parser; }
+        }
+
+        /// <summary>
+        /// Reference to the lexer.
+        /// </summary>
+        public ExpressionLexer Lexer
+        {
+            get { return this.lexer; }
+        }
+
+        /// <summary>
+        /// Create a <see cref="CountSegmentToken"/>.
+        /// </summary>
+        /// <param name="identifier">The name of this token, which in this case is always "$count".The name of this token, which in this case is always "$count".</param>

```

Comment:
>The name of this token, which in this case is always "$count" [](http://example.com/codeflow?start=99&length=61)

text seems to be repeated

Reply:
Removed the `identifier` entirely since we don't need it.